openmc/deplete/chain.py

```diff
@@ -454,70 +454,99 @@ def form_matrix(self, rates):
-        """Set the capture branching ratios
-
-        To provide a buffer around floating point precisions,
-        the sum of all branching ratios from a single parent
-        cannot be greater than 1.00001.
+    def set_branch_ratios(self, branch_ratios, reaction="(n,gamma)",
+                          strict=True, tolerance=1e-5):
+        """Set the branching ratios for a given reactions
 
         Parameters
         ----------
         branch_ratios : dict of {str: {str: float}}
             Capture branching ratios to be inserted.
             First layer keys are names of parent nuclides, e.g.
-            ``"Am241"``. The capture branching ratios for these
+            ``"Am241"``. The branching ratios for these
             parents will be modified. Corresponding values are
             dictionaries of ``{target: branching_ratio}``
-        strict : bool
-            If this evalutes to ``True``, then all parents and
-            products must exist in the :class:`Chain`. A
-            :class:`KeyError` will be raised at the first
-            nuclide that does not exist. Otherwise, print
-            a warning message for missing parents and/or
-            products.
+        reaction : str, optional
+            Reaction name like ``"(n,gamma)"`` [default], or
+            ``"(n, alpha)"``.
+        strict : bool, optional
+            Error control. If this evalutes to ``True``, then errors will
+            be raised if inconsistencies are found. Otherwise, warnings
+            will be raised for most issues.
+        tolerance : float, optional
+            Tolerance on the sum of all branching ratios for a
+            single parent. Will be checked with::
+
+                1 - tol < sum_br < 1 + tol
+
+        Raises
+        ------
+        IndexError
+            If no isotopes were found on the chain that have the requested
+            reaction
+        KeyError
+            If ``strict`` evaluates to ``False`` and a parent isotope in
+            ``branch_ratios`` does not exist on the chain
+        AttributeError
+            If ``strict`` evaluates to ``False`` and a parent isotope in
+            ``branch_ratios`` does not have the requested reaction
+        ValueError
+            If ``strict`` evalutes to ``False`` and the sum of one parents
+            branch ratios is outside  1 +/- ``tolerance``
 
         See Also
         --------
-        :meth:`get_capture_branches`
+        :meth:`get_branch_ratios`
         """
 
         # Store some useful information through the validation stage
 
         sums = {}
-        capt_ix_map = {}
+        rxn_ix_map = {}
         grounds = {}
 
+        tolerance = abs(tolerance)
+
         missing_parents = set()
         missing_products = {}
-        no_capture = set()
+        missing_reaction = set()
+        bad_sums = {}
+
+        # Secondary products, like alpha particles, should not be modified
+        secondary = "He4" if reaction == "(n,a)" else None
```

Comment:
Absolutely. I'll add that.

Yeah, we should probably include those. We can also add checks for those in `Chain.validate`. A good place to add those would be in `Chain.from_endf` and the CASL script, since these are the primary ways to generate a depletion chain "from scratch"